### ISSUE TITLE
Non-trivial tasklists parsed/rendered incorrectly

### ISSUE THREAD

**Issue by Turbo87** [kivikakk/comrak#558]
It looks like tasklists are currently only supported for plain text contents:

### Input

```md
- [ ] `c`
- [ ] [link](https://crates.io)
- [ ] [link](#anchor)
```

### Expected

- [ ] `c`
- [ ] [link](https://crates.io)
- [ ] [link](#anchor)

```html
<ul>
<li><input type="checkbox" disabled=""> <code>c</code></li>
<li><input type="checkbox" disabled=""> <a href="https://crates.io" rel="nofollow noopener noreferrer">link</a></li>
<li><input type="checkbox" disabled=""> <a href="#anchor" rel="nofollow noopener noreferrer">link</a></li>
</ul>
```

### Output

```html
<ul>
<li><input type="checkbox" disabled=""></li>
<li><input type="checkbox" disabled=""></li>
<li><input type="checkbox" disabled=""></li>
</ul>
```

### Related

- https://github.com/rust-lang/crates.io/pull/10913

**Comment by digitalmoksha** [comment]
Hmm, indeed @Turbo87 - thanks for opening this!

I bisected it down to the commit https://github.com/kivikakk/comrak/commit/1c55e16f4fe027d9de76250a30f7c8edd906d357. It looks like the `parent.detach();` on line 3084 is the culprit. Comment says it was added to fix the fuzz test `tests::fuzz::echaw9`.

**Comment by digitalmoksha** [comment]
It's only happening when something like `<code>` is at the start of the item. So ``- [ ] test `code` `` works but ``-[ ] `code` `` doesn't.

@kivikakk I haven't yet found the right combination of checks to have it do what you want. Maybe if the parent (which is a paragraph) has only one child node 🤔

**Comment by Turbo87** [comment]
I can confirm that https://github.com/kivikakk/comrak/pull/559 fixes the issue. thank you @digitalmoksha! ❤
